### PR TITLE
Allow to rename tags from the web interface.

### PR DESCRIPTION
**Pull request by shulard**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documentation | no
| Translation   | yes
| CHANGELOG.md  | no
| Fixed tickets | #720
| License       | MIT

This PR will add the ability to rename tags. As discussed in #720 the process to rename is tag is :

* Create a new tag with the new label;
* Attach the new tag to all the entries attached to the old one;
* Drop the old one.

We need to discuss further about UX implementation.

### CODE REVIEW

**Comment by shulard** [comment]
Hello,

I've finished to implement the tag renaming process. Actually it works only inside the "material" theme but it can be implemented easily in the other one.

It use a form to allow the user to edit the label. Then it's submitted as a POST request and handled with a Symfony form.

I've some questions about implementation details :

* I've added a class and some style on the form element, I hope it was the right place to do it 😄 ;
* I use a single RenameTagType form instance to generate all the rename tag form in the list view, is it ok for you ?
* I've added some translations in the fr / en dictionaries, do I need to add them on all the languages ?
* JavaScript interactions is actually defined in the `tags.html.twig` file and injected in the footer. I can't find the right file to add my functions, is it ok for you ?

**Comment by shulard** [comment]
Also, if I need to add some details in the documentation please tell me where !

File: src/Wallabag/CoreBundle/Controller/TagController.php [hunk 1]
```diff
@@ -130,4 +132,46 @@ public function showEntriesForTagAction(Tag $tag, $page, Request $request)
...
+            foreach ($entries as $entry) {
+                $this->get('wallabag_core.tags_assigner')->assignTagsToEntry(
+                    $entry,
+                    $form->get('label')->getData()
+                );
+            }
+
+            $em = $this->getDoctrine()->getManager();
+            $em->remove($tag);
+            $em->flush();
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Controller/TagController.php]: Be careful, a tag can be associated to entries from other users.
IMHO you shouldn't delete it from the database but rather deassociate it from each entry of the current logged in user.

File: src/Wallabag/CoreBundle/Resources/views/themes/material/Tag/tags.html.twig [hunk 2]
```diff
@@ -13,7 +13,16 @@
         <ul class="card-tag-labels">
             {% for tag in tags %}
                 <li title="{{tag.label}} ({{ tag.nbEntries }})" id="tag-{{ tag.id }}">
-                    <a href="{{ path('tag_entries', {'slug': tag.slug}) }}" class="card-tag-link">{{tag.label}} ({{ tag.nbEntries }})</a>
+                    <a href="{{ path('tag_entries', {'slug': tag.slug}) }}" class="card-tag-link">
+                        {{tag.label}}&nbsp;({{ tag.nbEntries }})
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Resources/views/themes/material/Tag/tags.html.twig]: I know it was already there like this, but I think `{{ tag.label }}` (adding spaces around) might be better

**Comment by j0k3r** [comment]
> I've added some translations in the fr / en dictionaries, do I need to add them on all the languages ?

You should add them in other langages too (add the english one and comment the line)

>  I can't find the right file to add my functions, is it ok for you ?

I think you can add them here: `app/Resources/static/themes/_global/index.js`

> Also, if I need to add some details in the documentation please tell me where !

About the doc, you can update that file https://github.com/wallabag/doc/blob/master/fr/user/tags.md

File: src/Wallabag/CoreBundle/Controller/TagController.php (hunk 1 above)
Comment (shulard) [reply src/Wallabag/CoreBundle/Controller/TagController.php]: Ok, I've updated the code to remove the tag from associated entries.

File: src/Wallabag/CoreBundle/Resources/views/themes/material/Tag/tags.html.twig (hunk 2 above)
Comment (shulard) [reply src/Wallabag/CoreBundle/Resources/views/themes/material/Tag/tags.html.twig]: Ok, done 😄.

**Comment by shulard** [comment]
Hello @j0k3r,

I've updated the translation files to add lines even if they are commented. I also moved my JavaScript code inside `index.js` file and added compatibility with the baggy theme.

I think that the code part is done for me 😄, I'll create another PR for the documentation on the specific repository.

**Comment by j0k3r** [comment]
Looks like tests are failing, you should fix them.
And also run `php bin/php-cs-fixer fix` to fix CS issue.

**Comment by shulard** [comment]
I've updated the CS issues and fixed the tests. Travis status will validate these updates.

**Comment by ewoks** [comment]
Hi @j0k3r , any progress on this? Looks like it is ready for merge, but stayed stale. Can you take a look? thanks

**Comment by ewoks** [comment]
friendly poke to @j0k3r please don't forget review

**Comment by Kdecherf** [comment]
@ewoks I will review it in the following days

**Comment by j0k3r** [comment]
Thanks @shulard :clap:

**Pull request merged**